pkg/aggregator/apis/aggregation/types.go

```diff
@@ -0,0 +1,136 @@
+package aggregation
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type DataPlaneService struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec   DataPlaneServiceSpec
+	Status DataPlaneServiceStatus
+}
+
+type DataPlaneServiceSpec struct {
```

Comment:
Is this something that can only ever be created by a plugin?  We don't want this to be a resource that could be applied generally.

In particular, for datasources -- I think we want a very strict/defined structure based on the plugin

Reply:
yep, it's a cluster level resource that should be defined by plugin authors. RBAC should prevent users from adding these